packages/webassemblyjs/test/index.js

```diff
@@ -5,14 +5,43 @@ const chai = require("chai");
 const { readFileSync } = require("fs");
 const path = require("path");
 const vm = require("vm");
+const { parse } = require("../../wast-parser/lib/");
+const assert = require("assert");
 
 const WebAssembly = require("../lib");
+const validate = require("../lib/compiler/validation/stack").default;
 
 function toArrayBuffer(buf) {
   return buf.buffer.slice(buf.byteOffset, buf.byteOffset + buf.byteLength);
 }
 
 describe("interpreter", () => {
+  describe("validation", () => {
+    const testSuites = glob.sync(
+      "packages/webassemblyjs/test/validation/**/module.wast"
+    );
+
+    testSuites.forEach(suite => {
+      describe(suite, () => {
+        const module = readFileSync(suite, "utf8");
+        const expectedErrors = readFileSync(
+          path.join(path.dirname(suite), "throws.txt"),
+          "utf8"
+        )
+          .split("\n")
+          .map(s => s.trim())
+          .filter(s => s.length > 0);
+
+        it("should give correct list of type errors", () => {
+          const ast = parse(module);
+          const errors = validate(ast);
+
+          assert.deepEqual(errors, expectedErrors);
+        });
+      });
+    });
+  });
+
```

Comment:
@xtuc What do you think of this way of running tests? See last two files in the diff for an exampl

Reply:
I'm not sure there is any value in having the test fixture, test data and expected output in three different places.

Reply:
Good suggestion, that makes sense for these small "explanatory" test cases :+1: I keep wondering if writing a parser for the spec tests `assert_invalid` construct is something we should do. I already started once but somehow regressed back to reformatting them by hand.

Reply:
We couldn't use the reformatter with this strategy though :thinking:

Reply:
That's a good point! The Try/catch + assert.equal logic already exists for this in the REPL.

Like:
```wat
(assert_invalid
  (module
    (func (result f32) (f32.convert_s/i32 (i64.const 0)))
  )
  "Expected type i32 but got i64"
)
```

Note that currently `make test` doesn't run the spec tests you will need to add this (maybe just your tests).

Reply:
I haven't looked into the repl in detail, but combined some more files into single test cases. The build still breaks due to some issue with the spec testsuite though. Locally my `testsuite` directory is empty is that correct? Even `master` doesnt pass `make-ci` locally for me, so I must be doing something wrong.

Reply:
You need to update your gitsubmodules

Reply:
I'll review it shortly, I think we have more than a MVP because it's already better than my first implementation.